playwright/pages/giftCardsPage.ts

```diff
@@ -51,6 +51,8 @@ export class GiftCardsPage extends BasePage {
 
   async clickIssueCardButton() {
     await this.issueCardButton.click();
+    await this.loader.waitFor({ state: "hidden" });
+    await this.giftCardDialog.waitFor({ state: "visible" });
```

Comment:
Suggestion: It seems like it would be a fit for a separate function as apart from clicking the button it also checks for modal mount